### DISCUSSION TITLE
GitHub GraphQL API Query For Branch Protection Rules: How to access the "Require a pull request before merging" setting

### DISCUSSION THREAD

**Discussion by ws-markb** [community/community#57406] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am trying to list all branch protection rules set up within our organisation, along with how each is set up. 

My current query looks like this:

```graphql
{
  organization(login: "MY_ORG") {
    repositories(first: 100) {
      nodes {
        name
        branchProtectionRules(first: 10) {
          nodes {
            pattern
            isAdminEnforced
            restrictsPushes
            blocksCreations
            allowsForcePushes
            allowsDeletions
            bypassPullRequestAllowances(first: 10) {
              totalCount
            }
            bypassForcePushAllowances(first: 10) {
              totalCount
            }
            pushAllowances(first: 10) {
              totalCount
              nodes {
                actor {
                  ... on App {
                    slug
                  }
                  ... on Team {
                    slug
                  }
                  ... on User {
                    login
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
```

This gives me _most_ of the information I need, but I am stuck finding the "Require a pull request before merging" setting value in the GraphQL schema.

Someone else is also having this same issue here: https://stackoverflow.com/questions/76384359/github-graphql-api-branch-protection-rule-how-do-i-get-require-a-pull-reque

Every other setting relating to branch protection rules seems to be available for querying, but there is just nothing I can see that corresponds to this setting. 

So, is there a way to get this setting in a GraphQL query please? Any help gratefully received!

**Comment by markashleybell** [comment, marked as answer]
I think I figured it out: 

https://stackoverflow.com/questions/76384359/github-graphql-api-branch-protection-rule-how-do-i-get-require-a-pull-reque/76422520#76422520
